packages/sysweb3-keyring/src/keyring-manager.ts

```diff
@@ -673,62 +676,79 @@ export class KeyringManager implements IKeyringManager {
     };
   };
 
-  // private async _createTrezorAccount(
-  //   coin: string,
-  //   slip44: string,
-  //   index: string,
-  //   label?: string
-  // ) {
-  //   const { accounts } = this.wallet;
-  //   const { descriptor: xpub, balance } =
-  //     await this.trezorSigner.getAccountInfo({ coin, slip44 });
-
-  //   const address = await this.trezorSigner.getAddress({ coin, slip44, index });
-
-  //   const accountAlreadyExists =
-  //     Object.values(
-  //       accounts[KeyringAccountType.Trezor] as IKeyringAccountState[]
-  //     ).some((account) => account.address === address) ||
-  //     Object.values(
-  //       accounts[KeyringAccountType.HDAccount] as IKeyringAccountState[]
-  //     ).some((account) => account.address === address) ||
-  //     Object.values(
-  //       accounts[KeyringAccountType.Imported] as IKeyringAccountState[]
-  //     ).some((account) => account.address === address);
-
-  //   if (accountAlreadyExists)
-  //     throw new Error(
-  //       'Account already exists, try again with another Private Key.'
-  //     );
-  //   if (!xpub || !balance || !address)
-  //     throw new Error(
-  //       'Something wrong happened. Please, try again or report it'
-  //     );
-
-  //   const id =
-  //     Object.values(accounts[KeyringAccountType.Trezor]).length < 1
-  //       ? 0
-  //       : Object.values(accounts[KeyringAccountType.Trezor]).length;
-
-  //   const trezorAccount = {
-  //     ...this.initialTrezorAccountState,
-  //     address,
-  //     label: label ? label : `Trezor ${id + 1}`,
-  //     id: id,
-  //     balances: {
-  //       syscoin: +balance,
-  //       ethereum: 0,
-  //     },
-  //     xprv: '',
-  //     xpub,
-  //     assets: {
-  //       syscoin: [],
-  //       ethereum: [],
-  //     },
-  //   } as IKeyringAccountState;
-
-  //   return trezorAccount;
-  // }
+  private async _createTrezorAccount(
+    coin: string,
+    slip44: string,
+    index: string,
+    label?: string
+  ) {
+    const { accounts, activeNetwork } = this.wallet;
+    const { descriptor: xpub, balance } =
+      await this.trezorSigner.getAccountInfo({ coin, slip44, index });
+    let ethPubKey = '';
+
+    const isEVM = coin === 'eth';
+
+    const address = isEVM
+      ? xpub
+      : this.syscoinTransaction.getAddress(xpub, +index, false);
+
+    if (isEVM) {
+      const response = await this.trezorSigner.getPublicKey({
+        coin,
```

Comment:
Doesn't getAccountInfo provides already a xpub? Depending on how you configure the derivation path (for Ethereum we get the address so 5 parameters do make sense) but for Bitcoin and utxo networks where we're interested in the xpub we can stop into the third level of the derivation path, something like this should work:
```js
async function getXpubFromAccountInfo() {
  // Request the account info
  const result = await TrezorConnect.getAccountInfo({
    coin: 'Bitcoin', // Use the desired coin name
    path: "m/49'/0'/0'", // Use the desired derivation path
  });

  if (result.success) {
    // Extract the xpub from the response
    const xpub = result.payload.descriptor;

    console.log('xpub:', xpub);
    return xpub;
  } else {
    console.error('Error:', result.payload.error);
    return null;
  }
}

getXpubFromAccountInfo();
```

Reply:
For EVM networks, the method TrezorConnect.getAccountInfo returns in descriptor the address of account requested with derivation path. In this case, we need to use an independent method to get EVM public key instead getAccountInfo.